### PR TITLE
get version from pyproject.toml

### PR DESCRIPTION
**Pull request by eniklas**
This removes `version.py` and gets the version from `pyproject.toml` so it's defined in a single place.

### CODE REVIEW

**Comment by eniklas** [comment]
@derek-keeler if you know off-hand how to fix the CI failures, go for it. I'm donking. Will figure it out tomorrow if you don't.

File: src/gamatrix/__main__.py [hunk 1]
```diff
@@ -445,9 +445,12 @@ def parse_cmdline(argv: List[str], docstr: str) -> Dict[str, Any]:
     )
     log = logging.getLogger()
 
+    version = pkg_resources.get_distribution("gamatrix").version
+
     opts = parse_cmdline(
         argv=sys.argv[1:],
         docstr=__doc__ if __doc__ is not None else "",
+        version=version,
```
Comment (eniklas) [inline src/gamatrix/__main__.py:453]: Without this new arg, `-v` was working fine but the tests failed with:

```
FAILED test/test_cmdline.py::test_new_cmdline_handling[No switches-commandline0-config_fields0-expected_values0] - NameError: name 'version' is not defined
FAILED test/test_cmdline.py::test_new_cmdline_handling[Assorted values all in one-commandline1-config_fields1-expected_values1] - NameError: name 'version' is not defined
FAILED test/test_cmdline.py::test_new_cmdline_handling[Only set the mode to server-commandline2-config_fields2-expected_values2] - NameError: name 'version' is not defined
FAILED test/test_cmdline.py::test_new_cmdline_handling[Allow the cache to update missing items.-commandline3-config_fields3-expected_values3] - NameError: name 'version' is not defined
```

Might be a better way to do it, I don't know docopt.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -36,7 +36,7 @@ jobs:
 
     - name: Analyze with mypy
       run: |
-        python -m mypy
+        python -m mypy --install-types --non-interactive
```
Comment (eniklas) [inline .github/workflows/ci.yml:39]: mypy tests were failing with:

```
src/gamatrix/__main__.py:29: error: Library stubs not installed for "pkg_resources"  [import]
src/gamatrix/__main__.py:29: note: Hint: "python3 -m pip install types-setuptools"
src/gamatrix/__main__.py:29: note: (or run "mypy --install-types" to install all missing stub packages)
src/gamatrix/__main__.py:29: note: See https://mypy.readthedocs.io/en/stable/running_mypy.html#missing-imports
```

There doesn't seem to be a way to add command-line args to mypy in the toml file, so I had to add them here.

Comment (derek-keeler) [reply .github/workflows/ci.yml:39]: This works just fine. I'll play around with this later and see if I can figure out how things are to be with `pyproject` and `mypy`. Also, I'll take a closer look at Ruff as it seems simpler. (There is also a VSCode plugin for it so local configuration should be a bit easier).

https://mypy.readthedocs.io/en/stable/config_file.html#example-pyproject-toml

File: src/gamatrix/__main__.py (hunk 1 above)
Comment (derek-keeler) [reply src/gamatrix/__main__.py:453]: Yes, this is correct - but I don't understand why it was working before. Maybe defining `VERSION` is picked up by `docopt`... will look at it later.

However, for your change, I believe this is the _correct_ way to do things. Tell `docopt` what the version is explicitly.

**Review by derek-keeler — COMMENTED** [review]
LGTM :ship:  it.

File: src/gamatrix/__main__.py (hunk 1 above)
Comment (eniklas) [reply src/gamatrix/__main__.py:453]: Before, `VERSION` was coming in from an import, so it makes sense it was available before basically as a global var.

**Pull request merged**